TLM/TLM/UI/Textures/SpeedLimitTextures.cs

```diff
@@ -20,196 +136,123 @@ public static class SpeedLimitTextures {
-        //     ushort trimIndex = Math.Max((ushort)5, index);
-        //     return textures[trimIndex];
-        // }
-
-        /// <summary>
-        /// Given the float speed, style and MPH option return a texture to render.
-        /// </summary>
-        /// <param name="spd">Speed to display.</param>
-        /// <returns>Texture to display.</returns>
-        public static Texture2D GetSpeedLimitTexture(SpeedValue spd,
-                                                     IDictionary<int, Texture2D> textureSource) {
-            // Select the source for the textures based on unit and the theme
-            bool mph = GlobalConfig.Instance.Main.DisplaySpeedLimitsMph;
-
-            // Round to nearest 5 MPH or nearest 10 km/h
-            ushort index = mph
-                               ? spd.ToMphRounded(MPH_STEP).Mph
-                               : spd.ToKmphRounded(KMPH_STEP).Kmph;
-
-            // Trim the index since 140 km/h / 90 MPH is the max sign we have
-            ushort upper = mph
-                               ? UPPER_MPH
-                               : UPPER_KMPH;
-
-            // Show unlimited if the speed cannot be represented by the available sign textures
-            if (index == 0 || index > upper) {
-                // Log._Debug($"Trimming speed={speedLimit} index={index} to {upper}");
-                return textureSource[0];
+            if (Themes.ContainsKey(selectedThemeName)) {
+                return Themes[selectedThemeName].Load();
             }
 
-            // Trim from below to not go below index 5 (5 kmph or 5 mph)
-            ushort trimIndex = Math.Max((ushort)5, index);
-            return textureSource[trimIndex];
+            GlobalConfig.Instance.Main.RoadSignTheme = GERMAN_KM_SIGNS;
+            Log.Info($"Road Sign theme changed to default ({GERMAN_KM_SIGNS})");
+            return Themes[GERMAN_KM_SIGNS].Load();
         }
 
-        /// <summary>For current display settings get texture dictionary with the road signs.</summary>
-        /// <returns>Texture source (loaded textures with keys matching speeds).</returns>
-        public static IDictionary<int, Texture2D> GetTextureSource() {
-            var configMain = GlobalConfig.Instance.Main;
-            // Select the source for the textures based on unit and the theme
-            bool mph = configMain.DisplaySpeedLimitsMph;
-
-            if (mph) {
-                switch (configMain.MphRoadSignStyle) {
-                    case MphSignStyle.SquareUS:
-                        return TexturesMphUS;
-                    case MphSignStyle.RoundUK:
-                        return TexturesMphUK;
-                    case MphSignStyle.RoundGerman:
-                        // Do nothing, this is the default above
-                        break;
-                }
-            }
+        // TODO: Split loading here into dynamic sections, static enforces everything to stay in this ctor
```

Comment:
why static ctor?

Reply:
The class is static.
The solution would be to make it a regular class and access via `SpeedLimitsTextures.Instance` a bit longer and slower.